Training_25/Program.cs

```diff
@@ -3,11 +3,43 @@
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
 // Program.cs
-// Program on main branch.
+// Program to check if a given password is strong or weak.
 // ------------------------------------------------------------------------------------------------
+using System.Text;
+using static System.Console;
+
 namespace Training_25;
+
 internal class Program {
-   static void Main (string[] args) {
-      Console.WriteLine ("Hello, World!");
+   static void Main () {
+      string reason = CheckPasswordStrength (ReadValidPassword ("Enter a password: "));
+      WriteLine (string.IsNullOrEmpty (reason)
+        ? "Password is strong." : $"Password is weak.\nReasons:\n{reason}");
+   }
+
+   // Continuously prompts the user until a non-empty password is entered.
+   static string ReadValidPassword (string prompt) {
+      while (true) {
+         Write (prompt);
+         string? input = ReadLine ();
+         if (!string.IsNullOrWhiteSpace (input)) return input.Trim ();
+         WriteLine ("Password cannot be empty.");
+      }
+   }
+
+   // Checks password strength based on length, uppercase, lowercase, digit and special characters.
+   // Returns reasons if it's weak.
+   static string CheckPasswordStrength (string password) {
+      var sb = new StringBuilder ();
+      if (password.Length < 6) sb.AppendLine ("- Password should contain at least 6 characters.");

```

Comment:
```suggestion
      if (password.Length < 6) sb.AppendLine ("- At least 6 characters.");
```
Instead of repeating `Password should contain`, this would make it concise.